Model/ProductData.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Metrilo\Analytics\Model;
+
+class ProductData
+{
+    public function __construct(
+        \Magento\Catalog\Model\ResourceModel\Product\CollectionFactory $productCollection,
+        \Magento\Store\Model\StoreManagerInterface $storeManager
+    ) {
+        $this->productCollection = $productCollection;
+        $this->storeManager      = $storeManager;
+    }
+
+    public function getProducts($storeId)
+    {
+        $productsArray = [];
+        $productOptions = '';
+        $products = $this->productCollection->create()->addAttributeToSelect('*')->addUrlRewrite()->addStoreFilter($storeId);
+
+        foreach ($products as $product) {
+            if ($product->getTypeId() == 'configurable' || $product->getTypeId() == 'bundle' || $product->getTypeId() == 'grouped') {
+                $productOptions = $this->getProductOptions($product, $storeId); // get product opitons for configurable/bundle/grouped product
+            }
+
+            if($product->isVisibleInSiteVisibility()) { // CHECK FOR PRODUCT CATALOG VISIBILITY
```

Comment:
What happens in the following scenario.
2017 - Order 333 has product X.
2018 - Product X is hidden from website.
How would we get information about the products in order 333?

Reply:
Information related to orders is extracted from different tables where product id for each product in order is being stored. To be precise order information is stored in sales_order and products related to that order are in sales_order_item where you can filter by order id and get all related product info like name, sku, id, etc. 

Summing it up, if order 333 has product X that means at some point that product was available for purchase (had stock quantity for example). If that quantity is depleted in time that same product will still be visible in frontend but with 'out of stock' status (thus making it unavailable for purchase or creating order with) but that product will still exist in catalog and will require importing to Metrilo. What isVisibleInSiteVisibility method is for is to ensure that imported products are part of the catalog, because product options are products themselves, but they are unavailable for reaching at any point except administration for configuration.